components/LoginForm/LoginForm.js

```diff
@@ -49,9 +49,12 @@ class LoginForm extends Component {
 
     try {
       const { user, token } = await login(values);
-      actions.setSubmitting(false);
-      actions.resetForm();
-      await onSuccess({ user, token });
+      const isRouted = await onSuccess({ token, user });
```

Comment:
If the onSuccess handler includes a route change, won’t this form unmount?!

Reply:
The `onSuccess` handler does include a route change currently, prior to this change even.

Reply:
Right, but I’m saying you’ve re-ordered things and the other methods won’t call because the form will unmount.

Reply:
I see. Is it necessarily a bad thing if the form will unmount? 

If we don't do this then we have the same issue with the double clicking.

Reply:
The only issue that I see is that the form wont reset in state on a route change and will be permanently set to submitting, but I can't think about any negative consequences. It's possible the form's state resets on unmount, in which case we could just ditch both methods anyways 🤷‍♂. Can you verify that logging in with the code that you currently have yields no run-time errors in the console?

Reply:
Sure, I have just now verified with Network Throttling feature on Chrome (Slow 3G) that this change works as expected and that there are no errors in the console during runtime.